src/UI/state/news/actions/news.action.ts

```diff
@@ -0,0 +1,121 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import {
+  fetchGraphQlData,
+  fetchGraphQlDataAuthenticated,
+} from '../../../api/api';
+import {
+  newsById,
+  upsertNewsMutation,
+  getAllNews,
+  getAllNewsIds,
+} from '../../../api/queries';
+import { News } from '../../state.types';
+import { AppState } from '../../store';
+import {
+  setCurrentNewsForEditing,
+  newsLoading,
+  setNewsItems,
+  setTopNewsItems,
+} from '../newsSlice';
+import { toast } from 'react-toastify';
+
+const sanitiseNews = (news: News): News => {
+  return {
+    ...news,
+    title: encodeURIComponent(news.title),
+    summary: encodeURIComponent(news.summary),
+    article: encodeURIComponent(news.article),
+  };
+};
+
+const unsanitiseNews = (news: News): News => {
+  return {
+    ...news,
+    title: decodeURIComponent(news.title),
+    summary: decodeURIComponent(news.summary),
+    article: decodeURIComponent(news.article),
+  };
+};
+
+export const upsertNews = createAsyncThunk(
+  'news/upsert',
+  async (news: News, { getState, dispatch }) => {
+    dispatch(newsLoading(true));
+    try {
+      const token = (getState() as AppState).auth.token;
+      const newNews = await fetchGraphQlDataAuthenticated(
+        upsertNewsMutation(sanitiseNews(news)),
+        token
+      );
+      if (news.id) {
+        toast.success('Updated news with id ' + newNews.data.createEditNews.id);
+      } else {
+        toast.success('Created news with id ' + newNews.data.createEditNews.id);
+      }
+      dispatch(
+        setCurrentNewsForEditing({
+          title: '',
+          summary: '',
+          article: '',
+          image: '',
+        })
+      );
+    } catch (e) {
+      console.error(e);
```

Comment:
Are we console erroring on errors? I don't mind either way, but we should be consistent.

Reply:
I'll add a mock-able logger for this